quizcomp/quiz.py

```diff
@@ -206,3 +205,14 @@ def create_variant(self, identifier = None, seed = None, all_questions = False):
         data['_skip_class_validations'] = [Quiz]
 
         return quizcomp.variant.Variant(**data)
+    
+    def total_points(self):
+        """
+        Calculate the total points for the quiz based on question groups.
+        """
+        total=0
+        
+        for group in self.groups:
+            if hasattr(group, 'points') and group.points:
+                total += group.pick_count * group.points      
```

Comment:
I would put a newline after this.
We want to group our code into "visual blocks" that generally do one thing to make it easier for readers.